### PR TITLE
Correct the opposite logic of "force_full"

### PR DESCRIPTION
**Pull request by zxiong**
It doesn't make sense that "force_full" enable fast-forward, so change
the opposite logic

ref #4810
https://pulp.plan.io/issues/4810

### CODE REVIEW

**Comment by zxiong** [comment]
Following up last comment from https://github.com/pulp/pulp/pull/3917, correct the opposite logic.

File: server/pulp/plugins/file/distributor.py [hunk 1]
```diff
@@ -70,7 +70,7 @@ def publish_repo(self, repo, publish_conduit, config):
         :return:                report describing the publish operation
         :rtype:                 pulp.plugins.model.PublishReport
         """
-        if config.get("force_full", False):
+        if not config.get("force_full", True):
```
Comment (ipanova) [inline server/pulp/plugins/file/distributor.py]: force_full flag if unspecified should be set to False.
Consider changing to:
```
if not config.get("force_full", False)
```

**Review by ipanova — CHANGES_REQUESTED** [review]
please address https://github.com/pulp/pulp/pull/3918/files#r283252715

**Comment by zxiong** [comment]
@ipanova, could you help to review it again?

**Review by ipanova — APPROVED** [review]

**Comment by dkliban** [comment]
ok test

**Comment by zxiong** [comment]
The failed unit tests are caused by test data which is return from "publish_conduit.get_units(criteria=criteria)" in the method "publish_repo_fast_forward", so the mock method "get_units(criteria=None)" in "pulp/devel/mock_distributor.py" is modified to add a condition to make up test data for "publish_repo_fast_forward" when the criteria likes "{'checksum': {"$in": ["sum1",...]}})"

Previously, the method "publish_repo_fast_forward" should not be covered by existing unit tests, so adding five test cases which are referring to test cases for the method "publish_repo".

Besides that, improves "publish_repo_fast_forward" to delete symlinks that pointing files were removed, and the change also is covered by new unit tests.

**Comment by zxiong** [comment]
@ipanova, @dkliban, @rohanpm, could you help to review the PR again?

**Comment by dkliban** [comment]
ok test

**Pull request merged**